src/stan/lang/ast/node/expression_type_vis_def.hpp

```diff
@@ -43,6 +43,16 @@ namespace stan {
       return expr_type(DOUBLE_T, 2);
     }
 
+    expr_type
+    expression_type_vis::operator()(const algebra_solver& e) const {
+      return expr_type(VECTOR_T, 0);
```

Comment:
Doesn't it return a real value?  All the doc seemed to indicate a `real[]` return, but this is a `vector` return.